controllers/datadogagent/feature/enabledefault/feature.go

```diff
@@ -184,14 +186,29 @@ func (f *defaultFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredC
 		f.customConfigAnnotationKey = object.GetChecksumAnnotationKey(string(feature.DefaultIDType))
 	}
 
-	return feature.RequiredComponents{
-		ClusterAgent: feature.RequiredComponent{
-			IsRequired: &trueValue,
-		},
-		Agent: feature.RequiredComponent{
-			IsRequired: &trueValue,
-		},
+	if f.otelAgentEnabled {
```

Comment:
is the plan to eventually add a separate feature for otel? if so, can we add a comment here indicating that this modification is temporary?